src/main/kotlin/uk/gov/dwp/dataworks/controllers/UserContainerController.kt

```diff
@@ -33,21 +36,23 @@ class UserContainerController {
         ApiResponse(responseCode = "400", description = "Failure, bad request")
     ])
     @PostMapping("/deployusercontainers")
-    fun launchTask(@RequestBody requestBody: Model){
-        if (existingUserServiceCheck.check(requestBody.userName, requestBody.ecsClusterName)){
+    fun launchTask(@RequestBody requestBody: Model): String{
+        if (requestBody.userName!=null){
+        if (existingUserServiceCheck.check(requestBody.userName, configService.getStringConfig(ConfigKey.ECS_CLUSTER_NAME))){
             logger.info("Redirecting user to running containers, as they exist")
-            return
+        } else {
+            taskDeploymentService.taskDefinitionWithOverride(
+                    requestBody.userName,
+                    requestBody.containerPort,
+                    requestBody.jupyterCpu,
+                    requestBody.jupyterMemory,
+                    requestBody.additionalPermissions
+            )
+            logger.info("Submitted request", "cluster_name" to configService.getStringConfig(ConfigKey.ECS_CLUSTER_NAME), "user_name" to requestBody.userName)
+        }
+            return "${configService.getStringConfig(ConfigKey.USER_CONTAINER_URL)}/${requestBody.userName}"
+        }else {
+            throw IllegalArgumentException("No userName found in Request")
```

Comment:
I am not sure what you mean by this, there is an if statement that checks for null value for requestBody.username, so it will just throw the exception, if it has not been provided.